edit/edit.go

```diff
@@ -461,6 +476,186 @@ func ContainsComments(expr build.Expr, str string) bool {
 	return false
 }
 
+// RemoveEmptySelectsAndConcatLists iterates the tree in order to turn
+// empty selects into empty lists and adjacent lists are concatenated
+func RemoveEmptySelectsAndConcatLists(e build.Expr) build.Expr {
+	switch e := e.(type) {
+	case *build.BinaryExpr:
+		if e.Op == "+" {
+			e.X = RemoveEmptySelectsAndConcatLists(e.X)
+			e.Y = RemoveEmptySelectsAndConcatLists(e.Y)
+
+			x, xIsList := e.X.(*build.ListExpr)
+			y, yIsList := e.Y.(*build.ListExpr)
+
+			if xIsList && yIsList {
+				return &build.ListExpr{List: append(x.List, y.List...)}
+			}
+
+			if xIsList && len(x.List) == 0 {
+				return e.Y
+			}
+
+			if yIsList && len(y.List) == 0 {
+				return e.X
+			}
+		}
+	case *build.CallExpr:
+		if x, ok := e.X.(*build.Ident); ok && x.Name == "select" {
+			if len(e.List) == 0 {
+				return &build.ListExpr{List: []build.Expr{}}
+			}
+
+			if dict, ok := e.List[0].(*build.DictExpr); ok {
+				for _, keyVal := range dict.List {
+					if keyVal, ok := keyVal.(*build.KeyValueExpr); ok {
+						val, ok := keyVal.Value.(*build.ListExpr)
+						if !ok || len(val.List) > 0 {
+							return e
+						}
+					} else {
+						return e
+					}
+				}
+
+				return &build.ListExpr{List: []build.Expr{}}
+			}
+		}
+	}
+
+	return e
+}
+
+// ComputeIntersection returns the intersection of the two lists given as parameters;
+// if the containing elements are not build.StringExpr, the result will be nil.
+func ComputeIntersection(list1, list2 []build.Expr) []build.Expr {
+	if list1 == nil || list2 == nil {
+		return nil
+	}
+
+	if len(list2) == 0 {
+		return []build.Expr{}
+	}
+
+	i := 0
+	for j, common := range list1 {
+		found := false
+		for _, elem := range list2 {
+			if str, ok := elem.(*build.StringExpr); ok {
+				if str.Value == common.(*build.StringExpr).Value {
```

Comment:
There's one more unchecked type cast, please check whether `common` is actually a `StringExpr`.

Reply:
Sorry, missed that one :D